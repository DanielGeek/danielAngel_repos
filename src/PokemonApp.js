import React, { useState } from 'react'
import { AddPokemon } from './components/AddPokemon';
import { PokemonGrid } from './components/PokemonGrid';

export const PokemonApp = ({ defaultPokemon = ['bulbasaur'] }) => {

  const [pokemones, setPokemones] = useState(defaultPokemon);

  return (
    <>
      <h2 className='header-title'>Listado de Pokemon</h2>
      <AddPokemon setPokemones={setPokemones} />

      <ol className='container'>
      {
        pokemones.map( pokemonNameOrId => (
            <PokemonGrid
                key={ pokemonNameOrId }
                pokemonNameOrId={pokemonNameOrId}
            />
        ))
      }
      </ol>
    </>
  )
}
