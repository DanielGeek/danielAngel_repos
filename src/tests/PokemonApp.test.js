import React from 'react';
import { shallow } from "enzyme"
import { PokemonApp } from "../PokemonApp"

describe('Pruebas en <PokemonApp />', () => {

  test('debe mostrarse correctamente', () => {

    const wrapper = shallow(<PokemonApp />)
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de mostrar una lista de pokemon', () => {

    const pokemones = ['Bulbasaur', 'Ivysaur'];
    const wrapper = shallow(<PokemonApp defaultPokemon={ pokemones } />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('PokemonGrid').length).toBe(pokemones.length);
  })
})