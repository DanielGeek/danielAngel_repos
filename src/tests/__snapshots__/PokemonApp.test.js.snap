// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PokemonApp /> debe de mostrar una lista de pokemon 1`] = `
<Fragment>
  <h2
    className="header-title"
  >
    Listado de Pokemon
  </h2>
  <AddPokemon
    setPokemones={[Function]}
  />
  <ol
    className="container"
  >
    <PokemonGrid
      key="Bulbasaur"
      pokemonNameOrId="Bulbasaur"
    />
    <PokemonGrid
      key="Ivysaur"
      pokemonNameOrId="Ivysaur"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <PokemonApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2
    className="header-title"
  >
    Listado de Pokemon
  </h2>
  <AddPokemon
    setPokemones={[Function]}
  />
  <ol
    className="container"
  >
    <PokemonGrid
      key="bulbasaur"
      pokemonNameOrId="bulbasaur"
    />
  </ol>
</Fragment>
`;
